tests/test_training.py

```diff
@@ -36,3 +37,31 @@ def test_training(max_steps: int = 100, threshold: float = 0.9):
                 break
 
         assert accuracy >= threshold, f"too small accuracy: {accuracy}"
+
+
+@pytest.mark.forked
+def test_decentralized_optimizer_step():
```

Comment:
I would like to have a more complex test here, but sure, we don't need it immediately.

Reply:
Resolution: will add a more complex test for a more advanced optimizer in a subsequent PR.